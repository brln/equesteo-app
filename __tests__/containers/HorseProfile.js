import configureStore from 'redux-mock-store'
import React from "react"
import {fromJS, Map} from "immutable"
import renderer from "react-test-renderer"
import { Provider } from 'react-redux'
import thunk from "redux-thunk"

import HorseProfileContainer from '../../containers/HorseProfile'


const mockStore = configureStore([thunk])

test('render', () => {
  const horseID = 'horseID1'
  const horse = fromJS({
    _id: horseID
  })

  const store = mockStore(fromJS({
    pouchRecords: {
      horsePhotos: {},
      horseUsers: {},
      horses: {
        horseID1: horse,
      }
    },
    localState: {}
  }))



  const tree = renderer.create(
    <Provider store={store}>
      <HorseProfileContainer
        horse={horse}
      />
    </Provider>)
  expect(tree.toJSON()).toMatchSnapshot()
})

