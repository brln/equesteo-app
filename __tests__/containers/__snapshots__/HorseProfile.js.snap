// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<RCTScrollView>
  <View>
    <View>
      <View
        style={
          Object {
            "borderBottomColor": null,
            "borderBottomWidth": null,
            "height": 647,
          }
        }
      >
        <View>
          <Image
            onError={[Function]}
            source={1}
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          />
          <View
            style={
              Object {
                "bottom": 30,
                "left": 10,
                "position": "absolute",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 25,
                  "textShadowColor": "black",
                  "textShadowOffset": Object {
                    "height": 1,
                    "width": 2,
                  },
                  "textShadowRadius": 7,
                }
              }
            >
              No Name
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          header={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderBottomWidth": null,
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 15,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 15,
              },
              Object {
                "padding": 5,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "paddingLeft": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#4b9ec1",
                }
              }
            >
              Description
            </Text>
          </View>
        </View>
        <View
          cardBody={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": -5,
                "paddingLeft": -5,
                "paddingRight": -5,
                "paddingTop": -5,
              },
              Object {
                "marginBottom": 30,
                "marginLeft": 20,
                "marginRight": 20,
              },
            ]
          }
        >
          <Text>
            nothing
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flexWrap": "nowrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            },
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <View
          header={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderBottomWidth": null,
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 15,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 15,
              },
              Object {
                "padding": 5,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "paddingLeft": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#4b9ec1",
                }
              }
            >
              Info
            </Text>
          </View>
        </View>
        <View
          cardBody={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": -5,
                "paddingLeft": -5,
                "paddingRight": -5,
                "paddingTop": -5,
              },
              Object {
                "flex": 1,
                "marginBottom": 30,
                "marginLeft": 20,
                "marginRight": 20,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "paddingRight": 10,
                      }
                    }
                  >
                    <Image
                      source={1}
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "height": null,
                            "resizeMode": "contain",
                            "width": null,
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "flex": 3,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "paddingBottom": 3,
                        }
                      }
                    >
                      <Text>
                        Birthday
                      </Text>
                    </View>
                    <Text
                      style={
                        Object {
                          "fontSize": 20,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      unknown
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "paddingRight": 10,
                      }
                    }
                  >
                    <Image
                      source={1}
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "height": null,
                            "resizeMode": "contain",
                            "width": null,
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "flex": 3,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "paddingBottom": 3,
                        }
                      }
                    >
                      <Text>
                        Breed
                      </Text>
                    </View>
                    <Text
                      style={
                        Object {
                          "fontSize": 20,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      none
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "paddingRight": 10,
                      }
                    }
                  >
                    <Image
                      source={1}
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "height": null,
                            "resizeMode": "contain",
                            "width": null,
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "flex": 3,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "paddingBottom": 3,
                        }
                      }
                    >
                      <Text>
                        Height
                      </Text>
                    </View>
                    <Text
                      style={
                        Object {
                          "fontSize": 20,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      unknown
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "paddingRight": 10,
                      }
                    }
                  >
                    <Image
                      source={1}
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "height": null,
                            "resizeMode": "contain",
                            "width": null,
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "flex": 3,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "paddingBottom": 3,
                        }
                      }
                    >
                      <Text>
                        Sex
                      </Text>
                    </View>
                    <Text
                      style={
                        Object {
                          "fontSize": 20,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      none
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
